Dockerfile

```diff
@@ -1,11 +1,16 @@
 # Step 1 - Build Front End Application
 FROM node:16 AS build-frontend
 
-# Copy the entire project - Webpack puts compiled assets into the Django folder
+# Copy the files required to install npm packages
 WORKDIR /app
-COPY . .
+COPY frontend/package.json frontend/package-lock.json frontend/.npmrc ./frontend/.nvmrc ./frontend/
+COPY frontend/bin/ ./frontend/bin/
+COPY frontend/env/ ./frontend/env/
 
 RUN cd frontend && npm ci --quiet --production
+
+# Copy the entire project - Webpack puts compiled assets into the Django folder
+COPY . .
```

Comment:
What's the reasoning behind splitting things up here?

Reply:
Same as above:
To align with Docker recommendations on how to optimize docker config files to take advantage of caching: https://docs.docker.com/build/cache/#order-your-layers. I enabled Docker layers caching but doesn't help if we copy the whole source code before installing the npm packages since any change to the source code triggers a rebuild of that layer.